### PR TITLE
init

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/sortStudents.ts [hunk 1]
```diff
@@ -1,16 +1,80 @@
...
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade',
+}
+
+export enum SortOrder {
+  ascending = 'asc',
+  descending = 'desc',
```
Comment (mariia-bila) [inline src/sortStudents.ts:19]: you can use literal type

File: src/sortStudents.ts [hunk 2]
```diff
@@ -1,16 +1,80 @@
...
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      newArrOfStudents.sort((s1, s2) => {
+        if (order === SortOrder.ascending) {
+          return s1[sortBy].localeCompare(s2[sortBy]);
+        }
+
+        return s2[sortBy].localeCompare(s1[sortBy]);
```
Comment (mariia-bila) [inline src/sortStudents.ts:41]: u can use ternary operator

File: src/sortStudents.ts [hunk 3]
```diff
@@ -1,16 +1,80 @@
...
+
+    case SortType.Married:
+      newArrOfStudents.sort((s1, s2) => {
+        if (order === SortOrder.ascending) {
+          return +s1[sortBy] - +(s2[sortBy]);
+        }
+
+        return +s2[sortBy] - +(s1[sortBy]);
+      });
+      break;
```
Comment (mariia-bila) [inline src/sortStudents.ts:63]: its possible to combine age and married

**Review by mariia-bila — APPROVED** [review]
Nice work!

File: src/sortStudents.ts (hunk 1 above)
Comment (ihor-jpeg) [reply src/sortStudents.ts:19]: agree with Mariia

File: src/sortStudents.ts [hunk 4]
```diff
@@ -1,16 +1,80 @@
...
+  const newArrOfStudents = [...students];
+
+  function gradeAverage(grades: number[]): number {
+    return grades.reduce((accum, grade) => accum + grade) / grades.length;
+  }
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      newArrOfStudents.sort((s1, s2) => {
```
Comment (ihor-jpeg) [inline src/sortStudents.ts:36]: use meaningful names. s1 and s2 is a no-no

File: src/sortStudents.ts [hunk 5]
```diff
@@ -1,16 +1,80 @@
...
+        if (order === SortOrder.ascending) {
+          return gradeAverage(s1.grades) - gradeAverage(s2.grades);
+        }
+
+        return gradeAverage(s2.grades) - gradeAverage(s1.grades);
+      });
+      break;
 
+    default:
+      break;
```
Comment (ihor-jpeg) [inline src/sortStudents.ts:76]: no break is needed for default if it is the last case. simply return the original array from this point

**Review by ihor-jpeg — CHANGES_REQUESTED** [review]
Good work! Let's fix some stuff

File: src/sortStudents.ts [hunk 6]
```diff
@@ -1,16 +1,59 @@
...
+    return grades.reduce((accum, grade) => accum + grade) / grades.length;
+  }
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      newArrOfStudents.sort((prevStudent, nextStudent) => (order === 'asc'
+        ? prevStudent[sortBy].localeCompare(nextStudent[sortBy])
+        : nextStudent[sortBy].localeCompare(prevStudent[sortBy])
+      ));
```
Comment (ihor-jpeg) [inline src/sortStudents.ts:36]:
```suggestion
      newArrOfStudents.sort((prevStudent, nextStudent) => (
        order === 'asc'
          ? prevStudent[sortBy].localeCompare(nextStudent[sortBy])
          : nextStudent[sortBy].localeCompare(prevStudent[sortBy])
      ));
```

File: src/sortStudents.ts [hunk 7]
```diff
@@ -1,16 +1,59 @@
...
+      break;
+
+    case SortType.Age:
+    case SortType.Married:
+      newArrOfStudents.sort((prevStudent, nextStudent) => (order === 'asc'
+        ? +prevStudent[sortBy] - +nextStudent[sortBy]
+        : +nextStudent[sortBy] - +prevStudent[sortBy]
+      ));
```
Comment (ihor-jpeg) [inline src/sortStudents.ts:44]:
```suggestion
    case SortType.Married:
      newArrOfStudents.sort((prevStudent, nextStudent) => (
        order === 'asc'
          ? +prevStudent[sortBy] - +nextStudent[sortBy]
          : +nextStudent[sortBy] - +prevStudent[sortBy]
      ));
```

File: src/sortStudents.ts [hunk 8]
```diff
@@ -1,16 +1,59 @@
...
+      break;
+
+    case SortType.AverageGrade:
+      newArrOfStudents.sort((prevStudent, nextStudent) => (order === 'asc'
+        ? gradeAverage(prevStudent.grades) - gradeAverage(nextStudent.grades)
+        : gradeAverage(nextStudent.grades) - gradeAverage(prevStudent.grades)
+      ));
+      break;
```
Comment (ihor-jpeg) [inline src/sortStudents.ts:52]:
```suggestion
    case SortType.AverageGrade:
      newArrOfStudents.sort((prevStudent, nextStudent) => (
        order === 'asc'
          ? gradeAverage(prevStudent.grades) - gradeAverage(nextStudent.grades)
          : gradeAverage(nextStudent.grades) - gradeAverage(prevStudent.grades)
      ));
      break;
```

**Review by ihor-jpeg — APPROVED** [review]
Good stuff! Be consistent with formatting

**Pull request closed without merging**